visualization/app/codeCharta/ui/searchBar/searchBar.component.ts

```diff
@@ -60,7 +78,41 @@ export class SearchBarController implements BlacklistSubscriber, SearchPatternSu
 	}
 
 	private isPatternBlacklisted(blacklist: BlacklistItem[], blacklistType: BlacklistType) {
```

Comment:
We should reuse `CodeMapHelper.getNodesByGitignorePath`. It is the actual search logic. Thus, we would not have to reimplement it and we could reuse the unit tests as well.

Reply:
the above method returns an array of the searched objects that are displayed .. it is true that the search and exclusion logic are very similar but for the exclusion we need a boolean value that we the set across all nodes. We had discussed the implementation with rueben and concluded this would be optimal to achieve this..